### ISSUE TITLE
Check use of Check mode.

### ISSUE THREAD

**Issue by centiqchris** [voidquark/el_patching#1]
The default value of `el_patching_check_mode` is false.  

This is used in your task here for example

```yaml
- name: Update all packages
  ansible.builtin.dnf:
    name: "*"
    state: latest
    exclude: "{{ el_patching_exclude_packages | default(omit) }}"
  check_mode: "{{ el_patching_check_mode }}"
  when: el_patching_method == "all"
```

I don't believe this has the result that you intended.

By setting check_mode to false, you cause the task to execute irrespective of whether --check is specified on the play - it does not mean that the task will not execute in check mode.  

https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_checkmode.html#enforcing-or-preventing-check-mode-on-tasks

What this will do is ensure that you will update your servers when --check is specified - and I don't think this is what you wanted.

**Comment by voidquark** [comment]
The definition of `check_mode: Can run in check_mode and return changed status prediction without modifying target`. This causes the task to be executed in `check_mode`. You can have a look here https://docs.ansible.com/ansible/latest/collections/ansible/builtin/dnf_module.html .

The task is working as expected. It only shows prediction if you set `el_patching_check_mode: true` in inventory. Nothing is updated on the target host.

**Comment by centiqchris** [comment]
I agree that if you set el_patching_check_mode to true in your inventory then those tasks will run in check mode when you execute the play - however that's not the intended usage for check mode.  If you leave el_patching_check_mode as false as per your default, if you were to include your role in as part of a playbook and run it with `ansible-playbook -i inventory.yml playbook.yml --check` it will update the packages regardless.  It would be the expectation that running a play with check mode enabled would not make these changes.

**Comment by voidquark** [comment]
Now I understand what you mean. Both ways are valid ( just different approaches). I released the new version `1.0.2` where I implemented your suggestion.

Thank you for your contribution. :+1:
